### PR TITLE
Include palette and transparency size in evaluations

### PR DESCRIPTION
**Pull request by andrews05**
This is super minor but I started it back in January (in relation to #485) and thought I might as well open a PR for it.

For very small data, including the palette size in evaluations can sometimes achieve a better result.

issue-52-02 master: 1379 bytes
issue-52-02 PR: 1171 bytes

issue-52-03 master: 1555 bytes
issue-52-03 PR: 1315 bytes

issue-52-06 master: 249 bytes
issue-52-06 PR: 231 bytes (apparently 32bpp can be smaller than 2bpp!!)

I also tested this on the images mentioned in #418 and found oxipng now gives a better overall result than leanify.

### CODE REVIEW

**Comment by andrews05** [comment]
On hold until #500 lands. I might make it account for tRNS also.

**Comment by andrews05** [comment]
Reopened with added logic for transparency, which can help with even smaller data such as issue-52-06.

**Review by shssoichiro — APPROVED** [review]

**Comment by ace-dent** [comment]
Great addition! I work on small icons (16x16px), so this is essential.

> issue-52-06 PR: 231 bytes (apparently 32bpp can be smaller than 2bpp!!)

Interesting result, but is a known phenomenon. See this [work](https://optipng.sourceforge.net/pngtech/8bpp.html) by the OptiPNG author :-)

**Comment by andrews05** [comment]
@ace-dent Nice, hope it’s helpful for you.
Note that oxipng will still never attempt to *increase* bpp, e.g. 4-bit to 8-bit or indexed to rgb. There’s always room for further improvements :)

**Pull request merged**
